src/app/cli/src/cli_value_parser.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::str::FromStr;
+
+use internal_error::InternalError;
+use opendatafabric::{
+    DatasetName,
+    DatasetNamePattern,
+    DatasetRef,
+    DatasetRefAny,
+    DatasetRefPattern,
+    DatasetRefRemote,
+    Multihash,
+    RepoName,
+};
+use thiserror::Error;
+use url::Url;
+
+#[derive(Clone)]
+pub struct DatasetPatternValidationRes {
+    pub pattern: DatasetNamePattern,
+    pub is_dataset_ref: bool,
+}
```

Comment:
I don't see where this type is used

Reply:
Sorry, I cleaned up this, but didnt push yet